frontend/app/routes/$lang+/_protected+/personal-information+/email-address+/edit.tsx

```diff
@@ -0,0 +1,197 @@
+  const personalInformationRouteHelpers = getPersonalInformationRouteHelpers();
+  const personalInformation = await personalInformationRouteHelpers.getPersonalInformation(userInfoToken, params, request, session);
+  const emailAddress = personalInformation.emailAddress;
+
+  const csrfToken = String(session.get('csrfToken'));
+
+  const t = await getFixedT(request, handle.i18nNamespaces);
+  const meta = { title: t('gcweb:meta.title.template', { title: t('personal-information:email-address.edit.page-title') }) };
+
+  instrumentationService.countHttpStatus('email-address.edit', 200);
+  return json({ csrfToken, meta, emailAddress });
+}
+
+export async function action({ context: { session }, params, request }: ActionFunctionArgs) {
+  const log = getLogger('email-address/edit');
+
+  const instrumentationService = getInstrumentationService();
+  const raoidcService = await getRaoidcService();
+  const t = await getFixedT(request, handle.i18nNamespaces);
+
+  await raoidcService.handleSessionValidation(request, session);
+
+  const formDataSchema = z
+    .object({
+      emailAddress: z.string(),
+      confirmEmailAddress: z.string(),
+    })
+    .superRefine((val, ctx) => {
+      if (validator.isEmpty(val.emailAddress)) {
+        ctx.addIssue({ code: z.ZodIssueCode.custom, message: t('email-address.edit.error-message.empty-email-address'), path: ['emailAddress'] });
+      } else if (!validator.isEmail(val.emailAddress)) {
+        ctx.addIssue({ code: z.ZodIssueCode.custom, message: t('email-address.edit.error-message.invalid-email-format'), path: ['emailAddress'] });
+      }
+
+      if (validator.isEmpty(val.confirmEmailAddress)) {
+        ctx.addIssue({ code: z.ZodIssueCode.custom, message: t('email-address.edit.error-message.empty-email-address'), path: ['confirmEmailAddress'] });
+      } else if (!validator.isEmail(val.confirmEmailAddress)) {
+        ctx.addIssue({ code: z.ZodIssueCode.custom, message: t('email-address.edit.error-message.invalid-email-format'), path: ['confirmEmailAddress'] });
+      } else if (val.emailAddress !== val.confirmEmailAddress) {
+        ctx.addIssue({ code: z.ZodIssueCode.custom, message: t('email-address.edit.error-message.email-match'), path: ['confirmEmailAddress'] });
+      }
+    });
+
+  const formData = await request.formData();
+  const expectedCsrfToken = String(session.get('csrfToken'));
+  const submittedCsrfToken = String(formData.get('_csrf'));
+
+  if (expectedCsrfToken !== submittedCsrfToken) {
+    log.warn('Invalid CSRF token detected; expected: [%s], submitted: [%s]', expectedCsrfToken, submittedCsrfToken);
+    throw new Response('Invalid CSRF token', { status: 400 });
+  }
+
+  const data = {
+    confirmEmailAddress: formData.get('confirmEmailAddress') ? String(formData.get('confirmEmailAddress') ?? '') : undefined,
+    emailAddress: formData.get('emailAddress') ? String(formData.get('emailAddress') ?? '') : undefined,
```

Comment:
Convention in the application is to set fields to `undefined` if the field is optional, which these two fields are not.

```suggestion
    confirmEmailAddress: String(formData.get('confirmEmailAddress') ?? ''),
    emailAddress: String(formData.get('emailAddress') ?? ''),
```

Reply:
That makes sense, though is the empty string acceptable for required fields?

Reply:
Our Zod validation schema would catch that scenario:
`emailAddress: z.string().trim().min(1, t('email-address.edit.error-message.empty-email-address'))`